pymeasure/instruments/teledyne/teledyne_oscilloscope.py

```diff
@@ -0,0 +1,1136 @@
+        value = value[:-2]
+
+    return float(value)
+
+
+def _intensity_validator(value, values):
+    """Validate the input of the intensity property (grid intensity and trace intensity).
+
+    :param value: input parameters as a 2-element tuple
+    :param values: allowed space for each parameter
+    """
+    if not isinstance(value, tuple):
+        raise ValueError('Input value {} of trigger_select should be a tuple'.format(value))
+    if len(value) != 2:
+        raise ValueError('Number of parameters {} different from 2'.format(len(value)))
+    for i in range(2):
+        strict_discrete_range(value=value[i], values=values[i], step=1)
+    return value
+
+
+class _ChunkResizer:
+    """The only purpose of this class is to resize the chunk size of the instrument adapter.
+
+    This is necessary when reading a big chunk of data from the oscilloscope like image dumps and
+    waveforms.
+
+    .. Note::
+        Only if the new chunk size is bigger than the current chunk size, it is resized.
+
+    """
+
+    def __init__(self, adapter, chunk_size):
+        """Just initialize the object attributes.
+
+        :param adapter: Adapter of the instrument. This is usually accessed through the
+                        Instrument::adapter attribute.
+        :param chunk_size: new chunk size (int).
+        """
+        self.adapter = adapter
+        self.old_chunk_size = None
+        self.new_chunk_size = int(chunk_size) if chunk_size else 0
+
+    def __enter__(self):
+        """Only resize the chunk size if the adapter support this feature."""
+        if (self.adapter.connection is not None
+                and hasattr(self.adapter.connection, "chunk_size")):
+            if self.new_chunk_size > self.adapter.connection.chunk_size:
+                self.old_chunk_size = self.adapter.connection.chunk_size
+                self.adapter.connection.chunk_size = self.new_chunk_size
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.old_chunk_size is not None:
+            self.adapter.connection.chunk_size = self.old_chunk_size
+
+
+class TeledyneOscilloscopeChannel(Channel, metaclass=ABCMeta):
+    """A base abstract class for channel on a :class:`TeledyneOscilloscope` device."""
+
+    _BOOLS = {True: "ON", False: "OFF"}
+
+    BANDWIDTH_LIMITS = ["OFF", "ON"]
+
+    TRIGGER_SLOPES = {"negative": "NEG", "positive": "POS"}
+
+    # Capture and split a reply like "RMS,281E-6" and "RMS,281E-6,OK"
+    # The third response item ("state"), is not present in all oscilloscopes
+    # For compatibility it is captured if it can, but ignored otherwise
+    _re_pava_response = re.compile(r"^\s*"
+                                   r"(?P<parameter>\w+),\s*"
+                                   r"(?P<value>[^,]*)\s*"
+                                   r"(?:,(?P<state>\w+)\s*)?$")
+
+    bwlimit = Instrument.control(
+        "BWL?", "BWL %s",
+        """Set the internal low-pass filter for all channels.
```

Comment:
Is it really all channels? The command includes only a single channel in its name (from `insert_id`) and the instrument has a bwlimit command as well, which sets all channels.

Mabye a wrong copy paste from the wrong position?